incubator/hnc/pkg/controllers/hierarchy_controller.go

```diff
@@ -341,6 +344,42 @@ func (r *HierarchyReconciler) syncChildren(log logr.Logger, inst *tenancy.Hierar
 	}
 }
 
+// syncCritConditions enqueues the children of a namespace if the existing critical conditions in the
+// namespace are gone or critical conditions are newly found.
+func (r *HierarchyReconciler) syncCritConditions(log logr.Logger, ns *forest.Namespace, oldCritExists bool) {
+	newCritExists := ns.HasCritCondition()
+
+	// Early exit if there's no need to enqueue relatives.
+	if oldCritExists == newCritExists {
+		return
+	}
+
+	msg := "found"
+	if oldCritExists == true {
+		msg = "gone"
+	}
+	log.Info("Critical conditions are " + msg)
+	r.enqueueAffected(log, "children of a namespace with critical conditions "+msg, ns.ChildNames()...)
```

Comment:
This message should be singular because it's printed once for every enqueued child: `"child of a namespace with ..."`

Also, should this be _children_ or _descendants_? If having `CRIT_ANCESTOR` doesn't qualify as being a critical condition, the children will get enqueued, will _not_ appear to have a critical condition themselves, and therefore will not enqueue _their_ children - right?

Reply:
Right, here I should enqueue all descendants, not just children.

Replaced the log msg to "descendant" and func call to ```ns.DescendantNames()```

Reply:
Right, here I should enqueue descendants, not just children.

I replaced the log msg to say "descendant" and func call to ```ns.DescendantNames()```.